microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -191,21 +192,43 @@ def generate_testpmd_command(
         #   --eth-peer=<port id>,<receiver peer MAC address> \
         #   --stats-period <display interval in seconds>
         nic_include_info = self.generate_testpmd_include(nic_to_include, vdev_id)
+
+        # dpdk can use multiple cores to service the
+        # number of queues and ports present. Get the amount to work with.
+        cores_available = self.node.tools[Lscpu].get_core_count()
+
+        # default is to use cores 0 and 1
+        cores_to_use = 2
+
+        # queue test forces multicore, use_cores can override it if needed.
         if txq or rxq:
+
+            # set number of queues to use for tx and rx
             assert_that(txq).described_as(
                 "TX queue value must be greater than 0 if txq is used"
             ).is_greater_than(0)
             assert_that(rxq).described_as(
                 "RX queue value must be greater than 0 if rxq is used"
             ).is_greater_than(0)
             extra_args += f" --txq={txq} --rxq={rxq}  --port-topology=chained "
-            cores_to_use = self.node.tools[Lscpu].get_core_count()
-            assert_that(cores_to_use).described_as(
-                f"DPDK requires a minimum of 8 cores, found {cores_to_use}"
-            ).is_greater_than(8)
-            core_args = f"-l 0-{cores_to_use-1}"
-        else:
-            core_args = "-l 0-1"
+
+            # use either as many cores as we can or one for each queue and port
+            cores_to_use = min(cores_available, cores_to_use + txq + rxq)
+
+        # force set number of cores if the override argument is present.
+        if use_cores:
+            cores_to_use = use_cores
```

Comment:
The two names `cores_to_use` and `use_cores` are close here.  Can they be named better?